### DISCUSSION TITLE
Is `symfony console` an alias for `php bin/console`?

### DISCUSSION THREAD

**Discussion by nbro** [symfony/symfony#54927] [Q&A] [answered]
It seems that `symfony console` is just an alias for `php bin/console` or simply `bin/console` because of the hash bang `#!/usr/bin/env php` at the top of that file). Is this correct?

Using `diff` to check their differences `diff <(symfony console) <(php bin/console)` returns nothing. 

I suppose there's no preference and we can just use whatever we want, but I wonder why there's this alias, as this may create confusion

**Comment by ghost** [comment]
No. `symfony` is a cli binary written in Golang. `php` is php binary. However you can use both to run `bin/console` script.

**Reply by nbro** [reply]
I didn't say that `symfony` and `php` are the same thing. I said that `symfony console` is an alias for `bin/console`. I know what `php` is. It seems you have not read my post carefully.

**Reply by ghost** [reply]
I'm sure I understood your question. You've been learning Symfony for few months and I've been working with it for 10 years.

**Reply by nbro** [reply]
What is my question then and how did you answer it exactly?

**Comment by kira0269** [comment]
I think you can find some explanations in here https://symfony.com/doc/current/setup/symfony_server.html. There is a warning starting by "When using the Symfony binary with php bin/console (symfony console ...), the binary will always use environment variables detected via Docker and will ignore local environment variables."
So I think the symfony binary will add some logic before calling the console.

**Reply by kira0269** [reply]
Also, be careful: you compared the outputs of both `symfony console` and `php bin/console`.
As described in the official documentation, Symfony CLI provides many features, including deployment, local symfony project management, etc...
`symfony console` will not expose more commands than `bin/console`, but symfony cli will act as a wrapper to interact with your local docker containers for example

**Comment by nbro** [comment, marked as answer]
Here's the answer: https://github.com/symfony-cli/symfony-cli/blob/main/main.go#L73. It seems that what the Symfony CLI is doing is calling `php` with `bin/console` if we specify `symfony console`

**Reply by kira0269** [reply]
Yes, this is what I told you in my answer. But keep in mind that it also get environnement variables from the docker
